krabs/krabs/schema_locator.hpp

```diff
@@ -32,28 +32,84 @@ namespace krabs {
     struct schema_key
     {
         guid      provider;
+
+        // Using a string_view for name so that keys can be constructed from
+        // EVENT_RECORD pointers without allocation. If the key instance is
+        // added to the cache, 'internalize_name' must be called first so that
+        // the name points to owned memory and doesn't dangle.
+        // Only events logged with the TraceLogger API will have a name set
+        // in the key because it's available as part of the EVENT_RECORD.
+        // Other events are uniquely distinguished by their event Id.
```

Comment:
Is there any code snippet or documentation that we can add a link to, to assert that the record name is only set for trace loggers?